### ISSUE TITLE
prove various things are propositions

### ISSUE THREAD

**Issue by emilyriehl** [rzk-lang/sHoTT#56]
We now have various logically equivalent definitions of is-prop [here](https://rzk-lang.github.io/sHoTT/hott/09-propositions.rzk/). It would be great to apply the various notions to prove that various types are propositions.

For instance, for any type `A` we can show that `is-contr A` implies `is-prop A`. A proof should be called `is-prop-is-contr`.

But also, for any type `A`, whether or not `is-contr A` holds, `is-prop (is-contr A)`, i.e., `is-contr A` is a proposition. The [agda-unimath](https://unimath.github.io/agda-unimath/foundation-core.contractible-types.html?highlight=contractible%20types%20are%20propositions#contractible-types-are-propositions) library calls this `is-property-is-contr`. 

Again for any `f : A -> B`, `is-equiv A B f` is a proposition. This might be called `is-property-is-equiv`.

There are lots of other types we should be able to prove are properties, in both the HoTT and sHoTT libraries, but let's start with just a few to settle on good style.

**Comment by floverity** [comment]
Thanks! I'm keen to work on this.

**Comment by fredrik-bakke** [comment]
> The [agda-unimath](https://unimath.github.io/agda-unimath/foundation-core.contractible-types.html?highlight=contractible%20types%20are%20propositions#contractible-types-are-propositions) library calls this `is-property-is-contr`.

Right! For context, this is because of the naming conflict with the implication that contractible types are propositions, which is named `is-prop-is-contr`.

**Comment by emilyriehl** [comment]
Hey @floverity just to let you know I could use `is-property-is-equiv` in particular ;)

**Comment by emilyriehl** [comment]
This issue is re-open and would be a good first issue. 

The sHoTT library contains some [closure properties of propositions](https://rzk-lang.github.io/sHoTT/hott/09-propositions.rzk/) that can be applied to prove that various things are propositions. We already have: 

- [x] [is-prop-is-contr-itself](https://rzk-lang.github.io/sHoTT/hott/09-propositions.rzk/#define:is-prop-is-contr-itself) proving that `is-contr A` is a proposition

- [x] [is-prop-is-prop](https://rzk-lang.github.io/sHoTT/hott/09-propositions.rzk/#define:is-prop-is-prop) proving that `is-prop A` is a proposition.

These should be renamed to `is-property-X` following Agda-Unimath. Then identify other properties in the library and prove similarly that they are propositions:

- [ ] is-property-is-equiv
- [ ] is-property-is-segal
- [ ] is-property-is-rezk
- [ ] is-property-is-iso
- [ ] is-property-is-discrete
- [ ] is-property-is-covariant
- [ ] is-property-is-various-adjunction-stuff, etc

**Comment by fredrik-bakke** [comment]
Artem is working on proving `is-prop (is-equiv f)`.

**Comment by marco-david** [comment]
Here is an (AI) proof of is-property-is-segal. What do you think?

```
Being a Segal type is a proposition.

```rzk
#def is-property-is-segal
  ( A : U)
  : is-prop (is-segal A)
  -- This proof was written by ChatGPT 5.6 Sol.
  := is-prop-fiberwise-prop5 funext A (\ _ → A) (\ _ _ → A)
    (\ x y _ → hom A x y) (\ _ y z _ → hom A y z)
    (\ x y z f g → is-contr (Σ (h : hom A x z) , (hom2 A x y z f g h)))
    (\ x y z f g → is-prop-is-contr-itself (weakfunext-funext funext)
      (Σ (h : hom A x z) , (hom2 A x y z f g h)))
```
